### PR TITLE
Add `--allow` option in `build`

### PR DESCRIPTION
**Pull request by yankay**
fixes: https://github.com/containerd/nerdctl/issues/2744

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
Thanks

File: docs/command-reference.md [hunk 1]
```diff
@@ -681,6 +681,7 @@ Flags:
   - :whale: `type=image,name=example.com/image,push=true`: Push to a registry (see [`buildctl build`](https://github.com/moby/buildkit/tree/v0.9.0#imageregistry) documentation)
 - :whale: `--progress=(auto|plain|tty)`: Set type of progress output (auto, plain, tty). Use plain to show container output
 - :whale: `--secret`: Secret file to expose to the build: id=mysecret,src=/local/secret
+- :whale: `--allow`: Allow extra privileged entitlement, e.g. network.host, security.insecure
```
Comment (AkihiroSuda) [inline docs/command-reference.md:684]: Maybe have a link to how to configure the entitlements
https://github.com/moby/buildkit/blob/master/docs/buildkitd.toml.md

File: cmd/nerdctl/builder_build.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ If Dockerfile is not present and -f is not specified, it will look for Container
 	buildCommand.Flags().StringP("output", "o", "", "Output destination (format: type=local,dest=path)")
 	buildCommand.Flags().String("progress", "auto", "Set type of progress output (auto, plain, tty). Use plain to show container output")
 	buildCommand.Flags().StringArray("secret", nil, "Secret file to expose to the build: id=mysecret,src=/local/secret")
+	buildCommand.Flags().StringArray("allow", nil, "Allow extra privileged entitlement, e.g. network.host, security.insecure")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/builder_build.go:54]: Can we have a shell completion?

File: docs/command-reference.md (hunk 1 above)
Comment (yankay) [reply docs/command-reference.md]: Thanks @AkihiroSuda 

The docs has been added :-)

File: cmd/nerdctl/builder_build.go (hunk 2 above)
Comment (yankay) [reply cmd/nerdctl/builder_build.go:54]: Thanks @AkihiroSuda
Shell completion has been added :-)

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**